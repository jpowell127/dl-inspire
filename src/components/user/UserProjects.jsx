import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { BEHANCE_GET_USER_PROJECTS_LAMBDA_URL } from "../../../utilities/constants";
import { Dimmer, Loader, Modal, Header, Item } from "semantic-ui-react";

export class UserProjects extends Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      projects: []
    };
  }

  componentDidMount() {
    this.setState({ loading: true });

    fetch(BEHANCE_GET_USER_PROJECTS_LAMBDA_URL, {
      method: "POST",
      body: JSON.stringify({ user: this.props.userId })
    })
      .then(res => res.json())
      .then(result => result.projects.slice(0, 5))
      .then(projects => this.setState({ projects, loading: false }))
      .catch(error => this.setState({ errors: error }));
  }

  render() {
    const { projects, loading } = this.state;
    return (
      <Fragment>
        {loading && (
          <Modal.Content>
            <Dimmer active>
              <Loader size="medium">Loading...</Loader>
            </Dimmer>
          </Modal.Content>
        )}
        {projects &&
          projects.length > 0 && (
            <Modal.Content>
              <Header as="h3" size="large">
                Projects
              </Header>
              <Item.Group divided>
                {projects.map(project => (
                  <Item key={project.id}>
                    <Item.Image size="tiny" src={project.covers[115]} />
                    <Item.Content>
                      <Item.Header>
                        <a href={project.url} target="_blank">
                          {project.name}
                        </a>
                      </Item.Header>
                    </Item.Content>
                  </Item>
                ))}
              </Item.Group>
            </Modal.Content>
          )}
      </Fragment>
    );
  }
}

UserProjects.propTypes = {
  userId: PropTypes.number.isRequired
};

export default UserProjects;
