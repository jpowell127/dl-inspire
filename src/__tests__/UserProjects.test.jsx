import React from "react";
import UserProjects from "../components/user/UserProjects";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<UserProjects />", () => {
  it("renders correctly", () => {
    const userId = 1;
    const wrapper = shallow(<UserProjects userId={userId} />);
    expect(wrapper).toMatchSnapshot();
  });
});
