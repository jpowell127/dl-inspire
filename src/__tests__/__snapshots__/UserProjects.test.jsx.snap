// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProjects /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProjects
    userId={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ModalContent>
          <Dimmer
            active={true}
          >
            <Loader
              size="medium"
            >
              Loading...
            </Loader>
          </Dimmer>
        </ModalContent>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Dimmer
            active={true}
          >
            <Loader
              size="medium"
            >
              Loading...
            </Loader>
          </Dimmer>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": <Loader
              size="medium"
            >
              Loading...
            </Loader>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Loading...",
              "size": "medium",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      false,
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ModalContent>
            <Dimmer
              active={true}
            >
              <Loader
                size="medium"
              >
                Loading...
              </Loader>
            </Dimmer>
          </ModalContent>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Dimmer
              active={true}
            >
              <Loader
                size="medium"
              >
                Loading...
              </Loader>
            </Dimmer>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "children": <Loader
                size="medium"
              >
                Loading...
              </Loader>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Loading...",
                "size": "medium",
              },
              "ref": null,
              "rendered": "Loading...",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        false,
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
